registry/server/tag_cleanup.go

```diff
@@ -0,0 +1,82 @@
+package server
+
+import (
+	"context"
+	regexp2 "regexp"
+	"strconv"
+	"time"
+)
+
+var tagMarkTimeKey = "tagMarkedForDeletionTSMin"
+var allowedTagStalenessMinutes = 60 // todo: pass this through via config
+
+// todo: should I handle errors, or just log?
+// todo: what calls this function? A timer? Should anything happen if it fails?
+// MarkForDeletion marks stored tags that have been stranded without an associated kafka resource.
+func(s *Server) MarkForDeletion(ctx context.Context) error {
+	markTimeMinutes := string(time.Now().Minute())
+
+	// Get all brokers from ZK.
+	brokers, errs := s.ZK.GetAllBrokerMeta(false)
+	if errs != nil {
+		return ErrFetchingBrokers
+	}
+
+	// Get all topics from ZK
+	rgex, _ := regexp2.Compile(".*")
```

Comment:
Since we know what the regex will be ahead of time, we can avoid compiling it at each request. There's a few ways to do it, one would be to assign it as a package-global var at the head of the file (it's not always ideal, but this is a case where we're not mutating the var). Another could be to slap it in as a field in the [Server](https://github.com/DataDog/kafka-kit/blob/c368b15888a0932fbbb1a5368f0a1d90fe1a51bc/registry/server/server.go#L33) struct.